src/Dfe.PlanTech.AzureFunctions/Functions/QueueReceiver.cs

```diff
@@ -80,16 +97,63 @@ private async Task ProcessMessage(ServiceBusReceivedMessage message, ServiceBusM
         }
         catch (Exception ex)
         {
-            Logger.LogError(ex, "Error processing message ID {Message}", message.MessageId);
-            await messageActions.DeadLetterMessageAsync(message, null, ex.Message, ex.StackTrace, cancellationToken);
+            if (ex is JsonException or CmsEventException)
+            {
+                Logger.LogError(ex, "Error processing message ID {Message}", message.MessageId);
+                await messageActions.DeadLetterMessageAsync(message, null, ex.Message, ex.StackTrace,
+                    cancellationToken);
+            }
+
+            var deliveryAttempts = 0;
```

Comment:
I think that it might be good if you could take out this redelivery code and make a simple class to handle the logic if possible. I doubt it would take much work, and it would be good to try and minimise tech debt if this functionality isn't urgently needed.

Obviously entirely up to you though!